books/projects/x86isa/machine/instructions/bit.lisp

```diff
@@ -777,4 +777,46 @@
                (x86 (write-*ip proc-mode temp-rip x86)))
               x86))
 
+(defmacro bswap (n src)
+  (if (equal n 0)
+    0
+    `(logapp 8 (logtail ,(* (1- n) 8) ,src)
+             (bswap ,(1- n) ,src))))
+
+(def-inst x86-bswap
+
+  ;; 0F C8 + rd: BSWAP r32
+  ;; 0F C8 + rd: BSWAP r64
+
+  :parents (two-byte-opcodes)
+
+  :returns (x86 x86p :hyp (x86p x86))
+  :guard-hints (("Goal" :in-theory (disable reg-indexp unsigned-byte-p)))
+
+  :body
+
+  (b* (;; Only 32 and 64 bit registers are allowed
+       ((the (integer 2 8) operand-size)
+        (select-operand-size
+         proc-mode nil rex-byte nil prefixes nil nil nil x86))
+
+       ;; SDM says undefined behavior for 16-bit register
```

Comment:
What is "SDM"? Can you spell it out?

Reply:
By SDM I meant Intel's Software Developer Manual. The acronym SDM is used in comments throughout the code, but I think it generally specifies "Intel's SDM." I'll update it to mention "Intel's SDM."